### ISSUE TITLE
Error parsing flow Generic types

### ISSUE THREAD

**Issue by jgrund** [babel/babel-eslint#280]
It seems like generics are no longer accounted for using `babel-eslint@6.0.0` and `eslint@2.5.1`.

```
// @flow

type returnXs = <T>(v:any, xs:T) => T;
```

yields:

```
  3:18  error    'T' is not defined                                              no-undef
  3:31  error    'T' is not defined                                              no-undef
  3:37  error    'T' is not defined                                              no-undef
```

**Comment by jedwards1211** [comment]
It was working for me with `babel-eslint@6.4.0` and `eslint@2.5.1`...I think the issue is actually in a newer version of some transitive dependency that's semver-compatible.

**Comment by jedwards1211** [comment]
@jgrund I was able to get back working sub-deps by...[forking npm](https://github.com/jedwards1211/npm/tree/time-machine)!

This might work for you:
```sh
cd ~
git clone git@github.com:jedwards1211/npm
cd npm
git checkout time-machine
cd <your-project>
rm -rf node_modules
~/npm/lib/npm.js install --time-machine=2016-05-27 # this date worked for me
```

**Comment by ryanspradlin** [comment]
Looks like the transitive dependency causing the issue is perhaps `babylon` -- adding it as an explicit dependency fixed at the older `6.7.0` version makes the incorrect ESLint errors go away for me.

**Comment by hzoo** [comment]
yeah the only relative dependency is babylon so we'll need to figure out what the change was

**Comment by MoOx** [comment]
Not sure if that's related, but I just updated to eslint 4+ (with latest flowtype plugin) and I am getting `'SyntheticEvent' is not defined` (but I didn't get this error before).

**Comment by kaicataldo** [comment]
Thank you for the issue. Now that [`@babel/eslint-parser` has been released](https://babeljs.io/blog/2020/07/13/the-state-of-babel-eslint), we are making this repository read-only. If this is a change you would still like to advocate for, please reopen this in the [babel/babel monorepo](https://github.com/babel/babel).
